### PR TITLE
cloud/leaderelection: fix and optimize leaderelection

### PR DESCRIPTION
**Pull request by Iceber**
Signed-off-by: Iceber Gu <wei.cai-nat@daocloud.io>

**What type of PR is this?**

/kind feature
/king bug

**What this PR does / why we need it**:
Add call `klog.Fatal` on trigger graceful shutdown failure.
Fix the error return.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: cloud/pkg/leaderelection/leaderelection.go [hunk 1]
```diff
@@ -45,47 +45,51 @@ func Run(cfg *config.CloudCoreConfig, readyzAdaptor *ReadyzAdaptor) {
...
 			// Trigger core.GracefulShutdown()
-			TriggerGracefulShutdown()
+			if err := TriggerGracefulShutdown(); err != nil {
+				klog.Fatalf("failed to gracefully terminate program: %v", err)
+			}
 		},
 	}
 
 	leaderElector, err := leaderelection.NewLeaderElector(*leaderElectionConfig)
 	// Set readyzAdaptor manually
```
Comment (GsssC) [inline cloud/pkg/leaderelection/leaderelection.go:87]: Move down as well : )

File: cloud/pkg/leaderelection/leaderelection.go [hunk 2]
```diff
@@ -130,69 +134,76 @@ func makeLeaderElectionConfig(config componentbaseconfig.LeaderElectionConfigura
...
-				// If the patch failure is due to update conflict, the necessary retransmission is performed
-				if i >= maxRetries {
-					klog.Errorf("updateMaxRetries(%d) has reached, failed to patching podReadinessGate: kubeedge.io/CloudCoreIsLeader because of update conflict", maxRetries)
-				}
-				continue
-			}
-			break
+	namespace := os.Getenv("CLOUDCORE_POD_NAMESPACE")
+	klog.Infof("CloudCore is running in pod %s/%s, try to patch PodReadinessGate", namespace, podname)
+	//TODO: use specific clients
```
Comment (GsssC) [inline cloud/pkg/leaderelection/leaderelection.go:144]: May not need this todo

**Review by GsssC — COMMENTED** [review]
Thx for contributions!  : ) lgtm overall

**Comment by Iceber** [comment]
@GsssC updated, Thanks

**Review by fisherxu — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**